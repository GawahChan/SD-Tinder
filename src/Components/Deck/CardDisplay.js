import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  CardContents,
  CardOverlay,
  CardContent,
  CardButtonContainer,
  CardSubtitleContainer,
  Container
} from "./style";

import {
  CardInfoButton,
  CardLikedButton,
  CardDislikedButton,
  CardSuperlikedButton
} from "../../Common/Styled/Icons.styles";
import { Header, Subtitle } from "../../Common/Styled/Typography.styles";

const CardDisplay = ({ stuntDouble, animateAction, action }) => {
  const [actionType, setActionType] = useState("");
  const [displayBio, setDisplayBio] = useState(false);

  const receiveAction = (id, actionType) => {
    animateAction(id, actionType);
    setActionType(actionType);
  };

  const toggleBio = () => setDisplayBio(!displayBio);

  const overlay = actionType ? actionType.toUpperCase() : action.toUpperCase();

  return (
    <CardContents actionColour={overlay} img={stuntDouble.img}>
      <CardOverlay actionColour={overlay}>{overlay}</CardOverlay>
      <CardContent>
        <Container>
          <Header>
            {stuntDouble.name}, {stuntDouble.age}
          </Header>
          <CardSubtitleContainer>
            <Subtitle>{stuntDouble.location}</Subtitle>
            <CardInfoButton onTap={toggleBio} />
          </CardSubtitleContainer>
          {displayBio && <Subtitle>{stuntDouble.bio}</Subtitle>}
        </Container>
        <Container>
          <CardButtonContainer>
            <CardDislikedButton
              onClick={() => receiveAction(stuntDouble.id, "disliked")}
            />
            <CardSuperlikedButton
              onClick={() => receiveAction(stuntDouble.id, "superliked")}
            />
            <CardLikedButton
              onClick={() => receiveAction(stuntDouble.id, "liked")}
            />
          </CardButtonContainer>
        </Container>
      </CardContent>
    </CardContents>
  );
};

CardDisplay.propTypes = {
  stuntDouble: PropTypes.object,
  animateAction: PropTypes.func,
  action: PropTypes.string
};
export default CardDisplay;
