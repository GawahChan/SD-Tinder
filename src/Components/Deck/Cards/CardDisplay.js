import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  CardContents,
  CardOverlay,
  CardContent,
  CardButtonContainer,
  CardSubtitleContainer,
  Container
} from "./style";

import {
  CardInfoButton,
  CardLikedButton,
  CardDislikedButton,
  CardSuperlikedButton
} from "../../../Common/Styled/Icons.styles";
import { Header, Subtitle } from "../../../Common/Styled/Typography.styles";

class CardDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      actionType: "",
      displayBio: false
    };
  }

  receiveAction = (id, actionType) => {
    this.props.animateAction(id, actionType);
    this.setState({ actionType });
  };

  toggleBio = () => this.setState({ displayBio: !this.state.displayBio });

  render() {
    const overlay = this.state.actionType
      ? this.state.actionType.toUpperCase()
      : this.props.action.toUpperCase();

    return (
      <CardContents actionColour={overlay} img={this.props.stuntDouble.img}>
        <CardOverlay actionColour={overlay}>{overlay}</CardOverlay>
        <CardContent>
          <Container>
            <Header>
              {this.props.stuntDouble.name}, {this.props.stuntDouble.age}
            </Header>
            <CardSubtitleContainer>
              <Subtitle>{this.props.stuntDouble.location}</Subtitle>
              <CardInfoButton onTap={this.toggleBio} />
            </CardSubtitleContainer>
            {this.state.displayBio && (
              <Subtitle>{this.props.stuntDouble.bio}</Subtitle>
            )}
          </Container>
          <Container>
            <CardButtonContainer>
              <CardDislikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "disliked")
                }
              />
              <CardSuperlikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "superliked")
                }
              />
              <CardLikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "liked")
                }
              />
            </CardButtonContainer>
          </Container>
        </CardContent>
      </CardContents>
    );
  }
}

CardDisplay.propTypes = {
  stuntDouble: PropTypes.object,
  animateAction: PropTypes.func,
  action: PropTypes.string
};
export default CardDisplay;
