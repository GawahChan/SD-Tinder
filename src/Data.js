const Data = [
    {id: 0, name: 'Mila Kunis', age: '30', location: 'LA', bio: 'Actress' }
];

export default Data;