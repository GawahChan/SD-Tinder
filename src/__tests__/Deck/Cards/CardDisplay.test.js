import React from 'react';
import { shallow } from 'enzyme';
import CardDisplay from '../../../Components/Deck/Cards/CardDisplay';
import Data from '../../../Data';

describe('CardDisplay Component', () => {
    let mockAnimateAction = jest.fn();

    let data = [Data[0]];
    let props = { stuntDouble: data, animateAction: mockAnimateAction, action: '' }
    const cardDisplay = shallow(<CardDisplay {...props} />)

    it('renders correctly', () => {
        expect(cardDisplay).toMatchSnapshot();
    });

    describe('receiveAction event handler', () => {
        afterEach(() => {
            mockAnimateAction.mockClear();
            cardDisplay.setState({ actionType: '' });
        });

        describe('user clicks disliked button', () => {
            let id = 0;
            let actionType = 'dislike';

            it('passes data to `animateAction` Prop, also sets `actionType` state to "disliked"', () => {

                cardDisplay.instance().receiveAction(id, actionType);

                expect(mockAnimateAction).toHaveBeenCalledTimes(1);
                expect(mockAnimateAction).toHaveBeenCalledWith(id, actionType);
                expect(cardDisplay.state('actionType')).toEqual(actionType)
            });
        });

        describe('user clicks superliked button', () => {
            let id = 0;
            let actionType = 'superlike';

            it('passes data to `animateAction` Prop, also sets `actionType` state to "superliked"', () => {

                cardDisplay.instance().receiveAction(id, actionType);

                expect(mockAnimateAction).toHaveBeenCalledTimes(1);
                expect(mockAnimateAction).toHaveBeenCalledWith(id, actionType);
                expect(cardDisplay.state('actionType')).toEqual(actionType)
            });
        });

        describe('user clicks liked button', () => {
            let id = 0;
            let actionType = 'liked';

            it('passes data to `animateAction` Prop, also sets `actionType` state to "liked"', () => {

                cardDisplay.instance().receiveAction(id, actionType);

                expect(mockAnimateAction).toHaveBeenCalledTimes(1);
                expect(mockAnimateAction).toHaveBeenCalledWith(id, actionType);
                expect(cardDisplay.state('actionType')).toEqual(actionType)
            });
        });
    });
});