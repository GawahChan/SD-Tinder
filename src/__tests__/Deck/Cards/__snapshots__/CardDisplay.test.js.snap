// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay Component renders correctly 1`] = `
<Styled(Component)
  actionColour=""
>
  <styled.h1
    actionColour=""
  />
  <styled.div>
    <CardDetails
      stuntDouble={
        Array [
          Object {
            "age": "28",
            "bio": "Recently finished stunt doubling for the movie 'Alita: Battle Angel' and am looking for a new role! Have stunt doubled for major films such as Wonder Woman (Gal Gadot), 'The Punisher' and 'Marvel's Agents of S.H.I.E.L.D'. If you're interested, swipe right and drop me a message!",
            "id": 0,
            "img": "Alicia-Vela-Bailey",
            "location": "LA, USA",
            "name": "Alicia Vela-Bailey",
          },
        ]
      }
    />
    <CardActions
      receiveAction={[Function]}
    />
  </styled.div>
</Styled(Component)>
`;
